src/eth_provider/provider.rs

```diff
@@ -572,6 +591,31 @@ where
         Ok(hash)
     }
 
+    async fn check_cached_balance(&self, signer: Address, transaction_value: U256) -> Result<(), EthApiError> {
```

Comment:
I think this is missing an important point: that when the transaction is pruned from the pending transactions database, we should update the cache with the new balance. 

Also, we should have a max cache size, otherwise this in memory cache will grow forever, so we should have a limit (~1000 accounts) and drop values in a FIFO way

Reply:
Fixed, I also added the size of the hashmap. Initially, I hadn't included it because every time a transaction is validated, the field is removed from the hashmap, so I didn't think it would grow much, but, it seems better this way.

Reply:
I just realized that the part about updating the balance when the transaction is removed from the pending transactions database can be removed, in my opinion. Because every time the transaction is validated with the `validate_transaction`, the balance is also removed from the cache. At this point, whether the transaction is removed from the database or not, the next time the user sends a new transaction, the balance is fetched again and the cache is updated. What do you think?